File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        arr=[]
        temp=""
        for i in s:
            if i!=' ':
                temp+=i
            else:
                if temp!="":
                    arr.append(temp)
                temp=""
        if temp!="":
            arr.append(temp)
        print(arr)
        start=0
        end=len(arr)-1
        while start<end:
            arr[start],arr[end]=arr[end],arr[start]
            start+=1
            end-=1
        return " ".join(arr)

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) {
        vector<string> str;
        
        string temp="";
        for(int i=0;i<s.size();i++){
            string sub = s.substr(i, 1);
            
            if(sub==" "){
                if(temp!=""){
                    bool res=temp!="";
                    cout<<res<<endl;
                    str.push_back(temp);
                }
                temp="";
                continue;
            }
            else{
                    temp+=sub;
            }
                
                
        }
        if(temp!="")
            str.push_back(temp);
        temp="";
        reverse(str.begin(),str.end());
        for(int i=0;i<str.size();i++){
            temp+=str[i];
            if(i<str.size()-1)
                temp+=" ";
        }
        return temp;
    }
};